Cargo.toml

```diff
@@ -43,15 +43,13 @@ tracing-limit = { path = "lib/tracing-limit" }
 # Tokio / Futures
 futures01 = { package = "futures", version = "0.1.25" }
 futures = { version = "0.3", default-features = false, features = ["compat", "io-compat"] }
-tokio01 = { package = "tokio", version = "0.1.22", features = ["io", "uds", "tcp", "rt-full", "experimental-tracing", "codec", "udp"], default-features = false }
+tokio01 = { package = "tokio", version = "0.1.22", features = ["io", "uds", "tcp", "rt-full", "experimental-tracing", "codec", "udp", "sync", "fs"], default-features = false }
```

Comment:
Does the new version of `pulsar` still depend on tokio 0.1? I'm curious why we need to add these.

Reply:
<details><summary>cargo tree -p pulsar</summary>

```
pulsar v1.0.0-alpha-2
├── bit-vec v0.6.1
├── bytes v0.5.4
│   └── serde v1.0.114
│       └── serde_derive v1.0.114
│           ├── proc-macro2 v1.0.18
│           │   └── unicode-xid v0.2.0
│           ├── quote v1.0.2
│           │   └── proc-macro2 v1.0.18 (*)
│           └── syn v1.0.33
│               ├── proc-macro2 v1.0.18 (*)
│               ├── quote v1.0.2 (*)
│               └── unicode-xid v0.2.0
├── chrono v0.4.11
│   ├── num-integer v0.1.42
│   │   └── num-traits v0.2.11
│   │       [build-dependencies]
│   │       └── autocfg v1.0.0
│   │   [build-dependencies]
│   │   └── autocfg v1.0.0
│   ├── num-traits v0.2.11 (*)
│   ├── serde v1.0.114 (*)
│   └── time v0.1.42
│       └── libc v0.2.69
├── crc v1.8.1
│   [build-dependencies]
│   └── build_const v0.2.1
├── futures v0.3.5
│   ├── futures-channel v0.3.5
│   │   ├── futures-core v0.3.5
│   │   └── futures-sink v0.3.5
│   ├── futures-core v0.3.5
│   ├── futures-executor v0.3.5
│   │   ├── futures-core v0.3.5
│   │   ├── futures-task v0.3.5
│   │   │   └── once_cell v1.4.0
│   │   └── futures-util v0.3.5
│   │       ├── futures v0.1.29
│   │       ├── futures-channel v0.3.5 (*)
│   │       ├── futures-core v0.3.5
│   │       ├── futures-io v0.3.5
│   │       ├── futures-macro v0.3.5
│   │       │   ├── proc-macro-hack v0.5.11
│   │       │   │   ├── proc-macro2 v1.0.18 (*)
│   │       │   │   ├── quote v1.0.2 (*)
│   │       │   │   └── syn v1.0.33 (*)
│   │       │   ├── proc-macro2 v1.0.18 (*)
│   │       │   ├── quote v1.0.2 (*)
│   │       │   └── syn v1.0.33 (*)
│   │       ├── futures-sink v0.3.5
│   │       ├── futures-task v0.3.5 (*)
│   │       ├── memchr v2.3.0
│   │       │   └── libc v0.2.69
│   │       ├── pin-project v0.4.22
│   │       │   └── pin-project-internal v0.4.22
│   │       │       ├── proc-macro2 v1.0.18 (*)
│   │       │       ├── quote v1.0.2 (*)
│   │       │       └── syn v1.0.33 (*)
│   │       ├── pin-utils v0.1.0
│   │       ├── proc-macro-hack v0.5.11 (*)
│   │       ├── proc-macro-nested v0.1.3
│   │       ├── slab v0.4.2
│   │       └── tokio-io v0.1.12
│   │           ├── bytes v0.4.12
│   │           │   ├── byteorder v1.3.2
│   │           │   ├── iovec v0.1.4
│   │           │   │   └── libc v0.2.69
│   │           │   └── serde v1.0.114 (*)
│   │           ├── futures v0.1.29
│   │           └── log v0.4.8
│   │               └── cfg-if v0.1.10
│   ├── futures-io v0.3.5
│   ├── futures-sink v0.3.5
│   ├── futures-task v0.3.5 (*)
│   └── futures-util v0.3.5 (*)
├── futures-io v0.3.5
├── futures-timer v3.0.2
├── log v0.4.8 (*)
├── native-tls v0.2.4
│   ├── log v0.4.8 (*)
│   ├── openssl v0.10.30
│   │   ├── bitflags v1.2.1
│   │   ├── cfg-if v0.1.10
│   │   ├── foreign-types v0.3.2
│   │   │   └── foreign-types-shared v0.1.1
│   │   ├── lazy_static v1.4.0
│   │   ├── libc v0.2.69
│   │   └── openssl-sys v0.9.58
│   │       └── libc v0.2.69
│   │       [build-dependencies]
│   │       ├── autocfg v1.0.0
│   │       ├── cc v1.0.50
│   │       │   └── jobserver v0.1.21
│   │       │       └── libc v0.2.69
│   │       ├── openssl-src v111.9.0+1.1.1g
│   │       │   └── cc v1.0.50 (*)
│   │       └── pkg-config v0.3.17
│   ├── openssl-probe v0.1.2
│   └── openssl-sys v0.9.58 (*)
├── nom v5.1.2
│   ├── lexical-core v0.7.4
│   │   ├── arrayvec v0.5.1
│   │   ├── bitflags v1.2.1
│   │   ├── cfg-if v0.1.10
│   │   ├── ryu v1.0.2
│   │   └── static_assertions v1.1.0
│   └── memchr v2.3.0 (*)
│   [build-dependencies]
│   └── version_check v0.9.1
├── pem v0.8.1
│   ├── base64 v0.12.0
│   ├── once_cell v1.4.0
│   └── regex v1.3.9
│       ├── aho-corasick v0.7.7
│       │   └── memchr v2.3.0 (*)
│       ├── memchr v2.3.0 (*)
│       ├── regex-syntax v0.6.18
│       └── thread_local v1.0.1
│           └── lazy_static v1.4.0
├── prost v0.6.1
│   ├── bytes v0.5.4 (*)
│   └── prost-derive v0.6.1
│       ├── anyhow v1.0.28
│       ├── itertools v0.8.2
│       │   └── either v1.5.3
│       ├── proc-macro2 v1.0.18 (*)
│       ├── quote v1.0.2 (*)
│       └── syn v1.0.33 (*)
├── prost-derive v0.6.1 (*)
├── rand v0.7.3
│   ├── getrandom v0.1.3
│   │   └── libc v0.2.69
│   ├── libc v0.2.69
│   ├── rand_chacha v0.2.1
│   │   ├── c2-chacha v0.2.3
│   │   │   └── ppv-lite86 v0.2.6
│   │   └── rand_core v0.5.1
│   │       └── getrandom v0.1.3 (*)
│   └── rand_core v0.5.1 (*)
├── regex v1.3.9 (*)
├── tokio v0.2.21
│   ├── bytes v0.5.4 (*)
│   ├── fnv v1.0.6
│   ├── futures-core v0.3.5
│   ├── iovec v0.1.4 (*)
│   ├── lazy_static v1.4.0
│   ├── libc v0.2.69
│   ├── memchr v2.3.0 (*)
│   ├── mio v0.6.21
│   │   ├── cfg-if v0.1.10
│   │   ├── iovec v0.1.4 (*)
│   │   ├── libc v0.2.69
│   │   ├── log v0.4.8 (*)
│   │   ├── net2 v0.2.33
│   │   │   ├── cfg-if v0.1.10
│   │   │   └── libc v0.2.69
│   │   └── slab v0.4.2
│   ├── mio-uds v0.6.7
│   │   ├── iovec v0.1.4 (*)
│   │   ├── libc v0.2.69
│   │   └── mio v0.6.21 (*)
│   ├── num_cpus v1.12.0
│   │   └── libc v0.2.69
│   ├── pin-project-lite v0.1.4
│   ├── signal-hook-registry v1.2.0
│   │   ├── arc-swap v0.4.4
│   │   └── libc v0.2.69
│   ├── slab v0.4.2
│   └── tokio-macros v0.2.5
│       ├── proc-macro2 v1.0.18 (*)
│       ├── quote v1.0.2 (*)
│       └── syn v1.0.33 (*)
├── tokio-native-tls v0.1.0
│   ├── native-tls v0.2.4 (*)
│   └── tokio v0.2.21 (*)
├── tokio-util v0.3.1
│   ├── bytes v0.5.4 (*)
│   ├── futures-core v0.3.5
│   ├── futures-io v0.3.5
│   ├── futures-sink v0.3.5
│   ├── log v0.4.8 (*)
│   ├── pin-project-lite v0.1.4
│   └── tokio v0.2.21 (*)
└── url v2.1.1
    ├── idna v0.2.0
    │   ├── matches v0.1.8
    │   ├── unicode-bidi v0.3.4
    │   │   └── matches v0.1.8
    │   └── unicode-normalization v0.1.12
    │       └── smallvec v1.2.0
    ├── matches v0.1.8
    └── percent-encoding v2.1.0
[build-dependencies]
└── prost-build v0.6.1
    ├── bytes v0.5.4 (*)
    ├── heck v0.3.1
    │   └── unicode-segmentation v1.6.0
    ├── itertools v0.8.2 (*)
    ├── log v0.4.8 (*)
    ├── multimap v0.8.1
    ├── petgraph v0.5.1
    │   ├── fixedbitset v0.2.0
    │   └── indexmap v1.3.1
    │       └── serde v1.0.114 (*)
    │       [build-dependencies]
    │       └── autocfg v1.0.0
    ├── prost v0.6.1 (*)
    ├── prost-types v0.6.1
    │   ├── bytes v0.5.4 (*)
    │   └── prost v0.6.1 (*)
    └── tempfile v3.1.0
        ├── cfg-if v0.1.10
        ├── libc v0.2.69
        ├── rand v0.7.3 (*)
        └── remove_dir_all v0.5.2
    [build-dependencies]
    └── which v3.1.1
        └── libc v0.2.69
```
</details>

There is tokio 0.1 in tree, but will be removed with futures 0.1. `sync` / `fs` features required by other crates...

<details><summary>log without these features</summary>

```
error[E0433]: failed to resolve: could not find `sync` in `tokio01`
   --> src/sinks/socket.rs:213:53
    |
213 |         let (mut close_tx, mut close_rx) = tokio01::sync::mpsc::unbounded_channel::<()>();
    |                                                     ^^^^ could not find `sync` in `tokio01`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `vector`.
```

```
error[E0432]: unresolved import `tokio01::fs`
 --> benches/files.rs:8:14
  |
8 | use tokio01::fs::OpenOptions;
  |              ^^ could not find `fs` in `tokio01`

error[E0432]: unresolved import `tokio01::fs`
 --> benches/files.rs:8:14
  |
8 | use tokio01::fs::OpenOptions;
  |              ^^ could not find `fs` in `tokio01`

error: aborting due to previous error
```
</details>

Reply:
The new `pulsar` version has no direct deps on tokio 0.1 or futures 0.1 (though it may be present somewhere deep in it's tree). It seems to me like this file could be refactored to use tokio 0.2 and futures 0.3 though, regardless of transitive dependencies.